truffle/src/main/ruby/core/thread.rb

```diff
@@ -16,6 +16,13 @@ def []=(symbol, value)
     __thread_local_variables[symbol] = value
   end
 
+  def thread_variable?(symbol)
+    __thread_local_variables.has_key? symbol
+  end
+
+  alias_method :thread_variable_get, :[]
+  alias_method :thread_variable_set, :[]=
```

Comment:
Yes I did figure out there was something wrong here - but actually I think our problem is that `Thread#[]` currently isn't fibre-local for us. It's stored in `ThreadLayout` and there's nothing similar in `FiberLayout`.